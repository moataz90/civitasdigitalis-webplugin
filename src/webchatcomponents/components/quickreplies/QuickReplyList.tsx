import * as React from 'react';
import { IPropsChildable } from '../../types/common-proptypes';
import { Message } from '../messages/Message';
import styled, { keyframes } from 'styled-components';
import { detect, BrowserInfo } from 'detect-browser';
import { Fragment } from 'react';

export interface IQuickReplyListProps extends IPropsChildable {
	children: React.ReactNode[];
	detectBrowser?: () => BrowserInfo;
}

const Animate = require('react-reveal/Zoom');

const FadeIn = keyframes`
0%   {opacity:0; transform:  translate(0px,15px); }
100% {opacity:1; transform:  translate(0px,0px);}

0%   {opacity:0; -moz-transform:  translate(0px,15px);}
100% {opacity:1; -moz-transform:  translate(0px,0px);}
0%   {opacity:0; -webkit-transform:  translate(0px,15px);}
100% {opacity:1; -webkit-transform:  translate(0px,0px);}
0%   {opacity:0; -o-transform:  translate(0px,15px)}
100% {opacity:1; -o-transform:  translate(0px,0px);}
0%   {opacity:0; -ms-transform:  translate(0px,15px);}
100% {opacity:1; -ms-transform:  translate(0px,0px);}
`;

const QuickReplyListAnimation = styled.div`
direction: rtl;
animation: ${FadeIn} ease 1s;
animation-iteration-count: 1;
transform-origin: 10% 10%;
animation-fill-mode:backwards; /*when the spec is finished*/
-webkit-animation: ${FadeIn} ease 1s;
-webkit-animation-iteration-count: 1;
-webkit-transform-origin: 10% 10%;
-webkit-animation-fill-mode:backwards; /*Chrome 16+, Safari 4+*/
-moz-animation: ${FadeIn} ease 1s;
-moz-animation-iteration-count: 1;
-moz-transform-origin: 10% 10%;
-moz-animation-fill-mode:backwards; /*FF 5+*/
-o-animation: ${FadeIn} ease 1s;
-o-animation-iteration-count: 1;
-o-transform-origin: 10% 10%;
-o-animation-fill-mode:backwards; /*Not implemented yet*/
-ms-animation: ${FadeIn} ease 1s;
-ms-animation-iteration-count: 1;
-ms-transform-origin: 10% 10%;
-ms-animation-fill-mode:backwards; /*IE 10+*/

opacity:0;
opacity: 1\9;
`;

const QuickReplyListComponent: React.StatelessComponent<IQuickReplyListProps> = (props) => {
	/* istanbul ignore next */
	const browser = (props.detectBrowser) ? props.detectBrowser() : detect();

	return (
		<Message isOwnMessage={true} isLastMessage={false}>
		{console.log(props.children.length)}
			{(browser.name === 'ie' || browser.name === 'safari' || browser.name === 'ios') ?
				<QuickReplyListAnimation >{props.children}</QuickReplyListAnimation> :
				/* istanbul ignore next */
				(props.children.length <= 1) ? <Animate bottom={true} ><Fragment >{props.children}</Fragment></Animate> :
					<Animate bottom={true} >{props.children}</Animate>

			}
		</Message>
	);
};

export const QuickReplyList = QuickReplyListComponent;