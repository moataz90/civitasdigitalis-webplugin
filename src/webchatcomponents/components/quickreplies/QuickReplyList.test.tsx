import * as React from 'react';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';
import { QuickReplyList } from './QuickReplyList';

import { BrowserInfo } from 'detect-browser';

// The following test cases are based on jest snapshot testing!
// For more information on how snapshot tests are working, see
// https://jest-bot.github.io/jest/docs/snapshot-testing.html

test('quick reply animation test IE', () => {
	
	const browser: BrowserInfo =  {
		name: 'ie',
		version: '1.0',
		os: 'windows'
	};

	const domTree = renderer.create(

		<QuickReplyList detectBrowser={(): BrowserInfo => {return browser; }}>
			<div>Some child</div>
			<div>Another child</div>
			</QuickReplyList>

	).toJSON();

	expect(domTree).toMatchSnapshot();
});

test('quick reply animation test safari', () => {

	const browser: BrowserInfo =  {
		name: 'safari',
		version: '1.0',
		os: 'windows'
	};

	const domTree = renderer.create(

		<QuickReplyList detectBrowser={(): BrowserInfo => {return browser; }}>
			<div>Some child</div>
			<div>Another child</div>
			</QuickReplyList>

	).toJSON();

	expect(domTree).toMatchSnapshot();
});

test('quick reply animation test other browser', () => {

	const browser: BrowserInfo =  {
		name: 'chrome',
		version: '1.0',
		os: 'windows'
	};

	const domTree = renderer.create(

		<QuickReplyList detectBrowser={(): BrowserInfo => {return browser; }}>
			<div>Some child</div>
			<div>Another child</div>
			</QuickReplyList>

	).toJSON();

	expect(domTree).toMatchSnapshot();
});

test('quick reply animation test ios', () => {

	const browser: BrowserInfo =  {
		name: 'ios',
		version: '1.0',
		os: 'windows'
	};

	const domTree = renderer.create(

		<QuickReplyList detectBrowser={(): BrowserInfo => {return browser; }}>
			<div>Some child</div>
			<div>Another child</div>
			</QuickReplyList>

	).toJSON();

	expect(domTree).toMatchSnapshot();
});