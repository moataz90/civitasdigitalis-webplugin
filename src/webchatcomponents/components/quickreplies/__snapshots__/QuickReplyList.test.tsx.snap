// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quick reply animation test IE 1`] = `
.c0.c0.c0 {
  width: 100%;
  position: relative;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding-right: 10px;
}

.c1 {
  direction: rtl;
  -webkit-animation: cKFngk ease 1s;
  animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -ms-transform-origin: 10% 10%;
  transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  animation-fill-mode: backwards;
  -webkit-animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  -moz-animation: cKFngk ease 1s;
  -moz-animation-iteration-count: 1;
  -moz-transform-origin: 10% 10%;
  -moz-animation-fill-mode: backwards;
  -o-animation: cKFngk ease 1s;
  -o-animation-iteration-count: 1;
  -o-transform-origin: 10% 10%;
  -o-animation-fill-mode: backwards;
  -ms-animation: cKFngk ease 1s;
  -ms-animation-iteration-count: 1;
  -ms-transform-origin: 10% 10%;
  -ms-animation-fill-mode: backwards;
  opacity: 0;
  opacity: 19;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div>
      Some child
    </div>
    <div>
      Another child
    </div>
  </div>
</div>
`;

exports[`quick reply animation test ios 1`] = `
.c0.c0.c0 {
  width: 100%;
  position: relative;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding-right: 10px;
}

.c1 {
  direction: rtl;
  -webkit-animation: cKFngk ease 1s;
  animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -ms-transform-origin: 10% 10%;
  transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  animation-fill-mode: backwards;
  -webkit-animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  -moz-animation: cKFngk ease 1s;
  -moz-animation-iteration-count: 1;
  -moz-transform-origin: 10% 10%;
  -moz-animation-fill-mode: backwards;
  -o-animation: cKFngk ease 1s;
  -o-animation-iteration-count: 1;
  -o-transform-origin: 10% 10%;
  -o-animation-fill-mode: backwards;
  -ms-animation: cKFngk ease 1s;
  -ms-animation-iteration-count: 1;
  -ms-transform-origin: 10% 10%;
  -ms-animation-fill-mode: backwards;
  opacity: 0;
  opacity: 19;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div>
      Some child
    </div>
    <div>
      Another child
    </div>
  </div>
</div>
`;

exports[`quick reply animation test other browser 1`] = `
.c0.c0.c0 {
  width: 100%;
  position: relative;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding-right: 10px;
}

<div
  className="c0"
>
  <div
    className="react-reveal"
    style={
      Object {
        "opacity": undefined,
      }
    }
  >
    Some child
  </div>
  <div
    className="react-reveal"
    style={
      Object {
        "opacity": undefined,
      }
    }
  >
    Another child
  </div>
</div>
`;

exports[`quick reply animation test safari 1`] = `
.c0.c0.c0 {
  width: 100%;
  position: relative;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding-right: 10px;
}

.c1 {
  direction: rtl;
  -webkit-animation: cKFngk ease 1s;
  animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -ms-transform-origin: 10% 10%;
  transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  animation-fill-mode: backwards;
  -webkit-animation: cKFngk ease 1s;
  -webkit-animation-iteration-count: 1;
  -webkit-transform-origin: 10% 10%;
  -webkit-animation-fill-mode: backwards;
  -moz-animation: cKFngk ease 1s;
  -moz-animation-iteration-count: 1;
  -moz-transform-origin: 10% 10%;
  -moz-animation-fill-mode: backwards;
  -o-animation: cKFngk ease 1s;
  -o-animation-iteration-count: 1;
  -o-transform-origin: 10% 10%;
  -o-animation-fill-mode: backwards;
  -ms-animation: cKFngk ease 1s;
  -ms-animation-iteration-count: 1;
  -ms-transform-origin: 10% 10%;
  -ms-animation-fill-mode: backwards;
  opacity: 0;
  opacity: 19;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div>
      Some child
    </div>
    <div>
      Another child
    </div>
  </div>
</div>
`;
